oxidation/libparsec/crates/client_types/src/wasm32/local_device.rs

```diff
@@ -0,0 +1,36 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 (eventually AGPL-3.0) 2016-present Scille SAS
+
+use libparsec_crypto::SecretKey;
+
+use crate::{DeviceFile, DeviceFileType, LocalDevice, LocalDeviceError, LocalDeviceResult};
+
+impl LocalDevice {
+    pub fn load_device_with_password(slug: &str, password: &str) -> LocalDeviceResult<Self> {
+        let window = web_sys::window().unwrap();
+        if let Ok(Some(storage)) = window.local_storage() {
+            if let Ok(Some(devices)) = storage.get_item("devices") {
+                let device = devices
+                    .split(':')
+                    .filter_map(|x| base64::decode(x).ok())
+                    .filter_map(|x| DeviceFile::load(&x).ok())
+                    .filter_map(|x| match x {
+                        DeviceFile::Password(x) => Some(x),
+                        _ => None,
+                    })
+                    .find(|x| x.slug.as_ref().map(|x| x.as_str()) == Some(slug))
+                    .ok_or_else(|| LocalDeviceError::NotFound { slug: slug.into() })?;
+
+                let key = SecretKey::from_password(password, &device.salt);
+                let data = key.decrypt(&device.ciphertext)?;
+
+                return LocalDevice::load(&data).map_err(|_| LocalDeviceError::Validation {
+                    ty: DeviceFileType::Password,
+                });
```

Comment:
This logic could be shared between `native` & `wasm32` logic on `load_device_with_password`

Reply:
True

Reply:
Done